go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -204,3 +215,20 @@ func (d DoltCommitFunc) Resolved() bool {
 func (d DoltCommitFunc) Children() []sql.Expression {
 	return d.children
 }
+
+// setHeadAndWorkingSessionRoot takes in a ctx and the new head hashstring and updates the session head and working hashes.
+func setHeadAndWorkingSessionRoot(ctx *sql.Context, headHashStr string) error {
+	key := ctx.GetCurrentDatabase() + sqle.HeadKeySuffix
+	dsess := sqle.DSessFromSess(ctx.Session)
+	hashType := sql.MustCreateString(query.Type_TEXT, 32, sql.Collation_ascii_bin)
+
+	return dsess.Set(ctx, key, hashType, headHashStr)
+}
+
+func setHeadSessionRoot(ctx *sql.Context, headHashStr string) error {
+	key := ctx.GetCurrentDatabase() + sqle.HeadKeySuffix
+	dsess := sqle.DSessFromSess(ctx.Session)
+	hashType := sql.MustCreateString(query.Type_TEXT, 32, sql.Collation_ascii_bin)
+
+	return dsess.Session.Set(ctx, key, hashType, headHashStr)
```

Comment:
If you're going to do this, make this a method on the dolt_session object, not a backdoor like this.

Reply:
SetSessionVarDirectly

And give it a good method doc on why you might want to do that